libs/savantboost/savantboost/deepstream/nvsurfaceptr.cpp

```diff
@@ -102,10 +107,20 @@ void DSCudaMemory::UnMapCudaPtr() {
     cudaError_t err;
     Npp8u* _gpu_frame = nullptr;
     int status;
+
     if (_surface != nullptr)
         if (_surface->memType == NVBUF_MEM_SURFACE_ARRAY){
             if (_EglImageMapBuffer){
-                if (NvBufSurfaceUnMapEglImage(_surface, 0) !=0) GST_ERROR("Error NvBufSurfaceUnMapEglImage");
+                NvBufSurfaceParams &surface = _surface->surfaceList[_batch_id];
+                err = cudaGraphicsUnregisterResource(_pResource);
```

Comment:
If this problem was due to a missing synchronization and cudaGraphicsUnregisterResource solves it, it is worth removing the warning and branch 'else' with an alternative way in DSCudaMemory::GetMapCudaPtr. and DSCudaMemory::UnMapCudaPtr. And all the time use only mapping through eglImage.

Reply:
@dorgun This solves the issue with crashes on long videos. I removed mapping through CPU memory.